### PR TITLE
Add FPort support

### LINKED ISSUE #2604: Implement FPort
BetaFlight and FrSky have developed [FPort](https://github.com/betaflight/betaflight/wiki/The-FrSky-FPort-Protocol), a new serial protocol that takes care of all the communications between RX and FC on a single wire: channels, telemetry, RSSI, MSP tunnelling... It can be added to existing X4R or XSR receivers by updating their firmware, and only needs a single s.port capable serial on the FC.

Given that most of us use two different ports for channels (sbus) and telemetry (s.port), this protocol would free a serial port, which would be great. Is it possible to merge it or it'd require a major overhaul of the API to do so?

### PR DESCRIPTION
**Pull request by fiam**
~Depends on #2902~ #2902 is now merged

FPort code is mostly imported from Betaflight. To make future RX
updates easier, a few more changes have been done to make the
RX/Telemetry code in BF and INAV a bit closer.

- Imported RSSI handling code from BF with some changes
- frsky_vfas_cell_voltage setting renamed to report_cell_voltage
- Added telemetry_halfduplex setting
- sbus_inversion renamed to serialrx_inverted, should apply to all
protocols eventually
- New MSP cmds MSP_TX_INFO/MSP_SET_TX_INFO now allow setting the RSSI
via MSP, thus RX_MSP doesn't need to use an extra channel anymore.

Fixes #2604

### CODE REVIEW

**Comment by fiam** [comment]
Just tested this on full duplex mode and it's working fine. As soon as anyone can give it a try, it's good to go. I'm gonna prepare a PR for the configurator. In the meantime, use `serialrx_provider` in the CLI. Note that you don't have to set any ports for telemetry.

**Comment by fiam** [comment]
Configurator PR at https://github.com/iNavFlight/inav-configurator/pull/386

**Comment by digitalentity** [comment]
@fiam which receiver did you use and how did you set up for full-duplex?

**Comment by fiam** [comment]
@digitalentity A custom one 🤣 . I tested it on the BF implementation just to make sure and it worked too. For an FrSky RX you'll need to set `serialrx_halfduplex = ON`. As far as I know, they are no official RXs using full duplex for now.

File: src/main/rx/rx.c [hunk 1]
```diff
@@ -435,6 +512,50 @@ void parseRcChannels(const char *input)
...
+        return;
+    }
+
+    rssi = newRssi;
+}
+
+#define RSSI_SAMPLE_COUNT 16
+#define RSSI_MAX_VALUE 1023
+
+void setRssiUnfiltered(uint16_t rssiValue, rssiSource_e source)
```
Comment (digitalentity) [inline src/main/rx/rx.c]: Since we are introducing this function it would be good to make other RSSI code to use it as well (channel/analog)

**Review by digitalentity — COMMENTED** [review]
This seems a direct grafting from Betaflight. I don't like the interrupt-driven implementation.

**Comment by fiam** [comment]
@digitalentity `fport.{c,h}` is mostly copied from BF, with just some minor adjustments. I agree that it's doing a bit too much work in the serial callback, but the proper fix is making `TASK_SERIAL` more flexible and call those callbacks from the main loop. That means that the drivers will be more or less oblivious of the change.

Regarding the RSSI, I'll use `setRssiUnfiltered()` to handle the other sources. I planned to do it on a separate PR, since the main focus of this one is FPort, but I'll add the commits to this PR once I clean them up.

**Comment by fiam** [comment]
Tested half duplex mode with custom RX, working fine too.

**Comment by fiam** [comment]
I've just realised we're using the FS bit in SBUS/FPort just to make the code go on the same path as if no signal at all was received (`rxSignalReceived` becomes false). Is that enough or should we do something better?

**Comment by digitalentity** [comment]
Ok, I'm going to merge this. 
@fiam regarding FS bit - I think current logic is ok.

**Comment by fiam** [comment]
@digitalentity Thanks!

**Comment by Pacman111178** [comment]
Hello Fiam. Have just tried it and it works great with a X4R but as soon as a GPS fix comes and the 4 GPS sensors are transmitted there are problems with the telemetry! 19 sensors does not seem to be able to do the telemetry!

**Comment by lo4wai** [comment]
I also tried fport with x9d+ and r-xsr with non-inverting pad. Rc channels are totally fine.  However the transmitter randomly reports sensor loss.  At telemetry screen, some sensors have square brackets around them, I guess they do not receive updates for a period of time.

**Comment by fiam** [comment]
I experienced this yesterday at the field too, but I haven’t been able to reproduce it consistently. Could you open an issue and write down the steps to reproduce it? Thanks!

**Comment by Pacman111178** [comment]
@fiam  So with me it comes as soon as the airplane has GPS fix; 15 sensors before everything was great after 19 sensors everything is running out of the oar!

https://youtu.be/jEmPv9sLV_k

**Pull request merged**